DevTools/Compiler/C++Compiler/LibCpp/Parser.cpp

```diff
@@ -88,6 +87,35 @@ Token Parser::consume(Token::Type expected_type)
     return tok;
 }
 
+NonnullRefPtr<Variable> Parser::create_unnamed_var(NonnullRefPtr<Type>& type)
+{
+    StringBuilder sb;
+
+    sb.appendf("_.%zu", m_unnamed_var_counter++);
+    auto name = sb.build();
+    return create_ast_node<Variable>(type, name);
+}
+
+template<typename... Args>
+BinaryExpression& Parser::maybe_correct_binop_tree(BinaryExpression& result, Expression& right, Args... args)
+{
+    if (right.is_binary_expression()) {
+        auto& bi_right = reinterpret_cast<BinaryExpression&>(right);
+
+        auto any_operator_match = [](auto const& expected, auto const&... rest) {
+            return ((expected == rest) || ...);
+        }(bi_right.binary_operation(), args...);
+
+        if (any_operator_match) {
+            auto right_left_node = bi_right.left();
+            result.set_right(right_left_node);
+            bi_right.set_left(result);
+            return bi_right;
```

Comment:
why is this happening anyway? the parser shouldn't create the wrong AST to begin with, no?
if there's an actual need for this, could you at least put a comment about what's being fixed here?

Reply:
It's a problem with all LL parsers. When parsing two operator with the same precedence, it will produce a tree with the branches inverted.
`i + j - k` will have a tree like `i + (j -k)`. This method transorm that tree into `(i + j) -k`.

I'll add a comment explaining that.

Reply:
Oh this is about + and -, I see.
For some reason I thought it was mixing up multiply and minus 🤔 
My bad~

Reply:
Comment has been added.